src/main/java/com/box/sdk/BoxTask.java

```diff
@@ -0,0 +1,281 @@
+        return addedAssignment.new Info(responseJSON);
+    }
+
+    /**
+     * Gets any assignments for this task.
+     * previous versions of their files.
+     * @return a list of previous file versions.
+     */
+    public List<BoxTaskAssignment.Info> getAssignments() {
+        URL url = GET_ASSIGNMENTS_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+
+        int totalCount = responseJSON.get("total_count").asInt();
+        List<BoxTaskAssignment.Info> assignments = new ArrayList<BoxTaskAssignment.Info>(totalCount);
+        JsonArray entries = responseJSON.get("entries").asArray();
+        for (JsonValue value : entries) {
+            JsonObject assignmentJSON = value.asObject();
+            BoxTaskAssignment assignment = new BoxTaskAssignment(this.getAPI(), assignmentJSON.get("id").asString());
+            BoxTaskAssignment.Info info = assignment.new Info(assignmentJSON);
+            assignments.add(info);
+        }
+
+        return assignments;
+    }
+
+    /**
+     * Gets information about this task.
+     * @return info about this task.
+     */
+    public Info getInfo() {
+        URL url = TASK_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new Info(responseJSON);
+    }
+
+    /**
+     * Updates the information about this task with any info fields that have been modified locally.
+     *
+     * <p>The only fields that will be updated are the ones that have been modified locally. For example, the following
+     * code won't update any information (or even send a network request) since none of the info's fields were
+     * changed:</p>
+     *
+     * <pre>BoxTask task = new BoxTask(api, id);
+     *BoxTask.Info info = task.getInfo();
+     *task.updateInfo(info);</pre>
+     *
+     * @param info the updated info.
+     */
+    public void updateInfo(BoxTask.Info info) {
+        URL url = TASK_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxJSONRequest request = new BoxJSONRequest(this.getAPI(), url, "PUT");
+        request.setBody(info.getPendingChanges());
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
+        info.update(jsonObject);
+    }
+
+    /**
+     * Contains information about a BoxTask.
+     */
+    public class Info extends BoxResource.Info {
+        private BoxFile.Info item;
+        private Date dueAt;
+        private String action;
```

Comment:
Instead of string, this would be better off as an Enum. The API does not allow free flow string here. Currently, the only allowed value is "review". As more values are allowed in future we can update the enum appropriately.